Sources_v3/StreamChatUI/ChatChannel/MessageComposer/MessageComposerViewController.swift

```diff
@@ -421,16 +423,28 @@ open class MessageComposerViewController<ExtraData: ExtraDataTypes>: ViewControl
         )
     }
 
-    func promptMentions(for text: String) {
+    private func promptMentions(for text: String?) {
         userSuggestionSearchController.search(term: text)
-
         showSuggestionsViewController(
             for: .mention,
-            onSelectItem: { [weak self] _ in
-                print(text)
-                print(self?.textView.text)
-                self?.textView.text = self?.textView.text.appending("@ \(text)")
-                self?.state = .initial
+            onSelectItem: { [weak self, textView = self.textView] userIndex in
+                guard let self = self else { return }
+
+                let user = self.userSuggestionSearchController.users[userIndex]
+                let cursorPositon = textView.selectedRange
+
+                let atRange = (textView.textStorage.string as NSString)
+                    .rangeOfCharacter(
+                        from: CharacterSet(charactersIn: "@"),
+                        options: .backwards,
+                        range: NSRange(location: 0, length: cursorPositon.location)
+                    )
+
+                textView.textStorage.replaceCharacters(
+                    in: NSRange(location: atRange.location, length: cursorPositon.location - atRange.location),
+                    with: "@\(user.id)"
+                )
```

Comment:
After you replaced mention with id, move cursor to the end of `user.id` and add 1 space character.

Reply:
Thanks! :) Done